### PR TITLE
`stringify`: fix arrayFormat comma with empty array/objects

### LINKED ISSUE #310: arrayformat as comma with empty array or objects
```javascript
const options = {
  encode: false,
  strictNullHandling: true,
  arrayFormat: 'comma',
  allowDots: true,
  addQueryPrefix: true
};

const params = {
  foo: [],
  bar: [null],
  baz: ['abc'],
};

// with arrayFormat: 'comma'
qs.stringify(params, options);
// = ?foo=&bar=&baz=abc

// with arrayFormat: 'brackets'
// = ?bar[]&baz[]=abc
```

`stringify` will produce `?foo=&bar=&baz=abc`. But using `brackets`, `indices`, `repeat` as array format will not concat `foo` cause it's an empty array. `comma` is the only one which keep foo. Is it a bug or a conception choice ?

In `lib/stringify.js`, line 67, maybe you could test if it's an array with values ?

### PR DESCRIPTION
**Pull request by daggerjames**
Fixed #310 

continued from PR #317

### CODE REVIEW

**Comment by ljharb** [comment]
Why did you open a second PR? That creates a **permanent** pollution in the repo refs that can never be removed.

Please never ever make a duplicate PR.

File: .gitignore [hunk 1]
```diff
@@ -10,3 +10,6 @@ dist/*
 yarn.lock
 package-lock.json
 npm-shrinkwrap.json
+
+# IDE related
+.idea
```
Comment (ljharb) [inline .gitignore]: revert this; your IDE files belong in your personal global gitignore, not littered in every project you touch

File: dist/qs.js [hunk 2]
```diff
@@ -243,7 +243,7 @@ var normalizeParseOptions = function normalizeParseOptions(opts) {
     }
 
     if (typeof opts.charset !== 'undefined' && opts.charset !== 'utf-8' && opts.charset !== 'iso-8859-1') {
-        throw new Error('The charset option must be either utf-8, iso-8859-1, or undefined');
+        throw new TypeError('The charset option must be either utf-8, iso-8859-1, or undefined');
```
Comment (ljharb) [inline dist/qs.js]: please revert all changes to this file; it only gets updated as part of a release

**Comment by ljharb** [comment]
@daggerjames hey, sorry for the delay here :-)

I've rebased this; the first commit is awesome and good to go; for the second one, 3 of your new tests are failing - and it *seems* like the tests are correct. Can you take a look?

**Comment by daggerjames** [comment]
fixed all tests. Seems logical error with previous approach.
could you please rebase the branch for me? thanks. I'm still unfamiliar with those rebase things and might lose it like what I did last time

Thanks~

**Comment by jonathansamines** [comment]
Hi @ljharb, sorry to bother you, but were these changes published to npm? I can still reproduce the original behavior from the latest published version. 

See this [failed test](https://runkit.com/jonathansamines/querystring-comma-empty-arrays) (extracted from this pull request)

**Comment by ljharb** [comment]
No; this hasn’t been published yet.

**Pull request merged**